dataclients/kubernetes/clusterstate.go

```diff
@@ -49,6 +55,8 @@ func (state *clusterState) GetEndpointsByService(namespace, name, protocol strin
 		TargetPort: servicePort.TargetPort.String(),
 	}
 
+	state.mu.Lock()
+	defer state.mu.Unlock()
```

Comment:
Ideally we need to get rid of `state.cachedEndpoints` (or prepupulate it during cluster state construction) to make cluster state read-only

Reply:
https://github.com/zalando/skipper/issues/2079